User_Story#1/SortingApp/BusinessLayer/Utilities/DataWriter.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+
+namespace BusinessLayer.Writer
+{
+    /// <summary>
+    /// Provides functionality that writes content to file.
+    /// </summary>
+    /// <owner>Anton Petrenko</owner>
+    public class DataWriter<T> where T : IComparable
+    {
+        /// <summary>
+        /// Gets or sets the path.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        /// <returns>The path.</returns>
+        public string Path { get; set; }
+
+        /// <summary>
+        /// Writes the collection by specified path.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        public void WriteContent(T[] collection)
+        {
+            if(collection is null || !collection.Any())
+            throw new ArgumentNullException(nameof(collection));
+            try
+            {
+                string temp = string.Empty;
+                using (var writer = new StreamWriter(Path))
+                {
+                    foreach (var value in collection)
```

Comment:
I believe that for should be excluded from try statement.
In that case you could get exception that doesn't belong to IO but you will think otherwise.

Reply:
Please explain.